### PR TITLE
1.6 for 1.7 bis

### PR DESCRIPTION
**Pull request by lienboe**

### CODE REVIEW

File: myocamlbuild.ml [hunk 1]
```diff
@@ -20,7 +20,7 @@ let errors = "+A-4-6-27-34-44"
  *
  * is disabled.
  *)
-let allow_fragile_pattern_matches = [ ]
+let allow_fragile_pattern_matches = ["logger_macro.ml";"src/tools/file_system.ml"; "src/node/node_cfg.ml"; "src/tools/server.ml";"src/node/catchup.ml";"src/tlog/tlc2.ml"; "src/msg/tcp_messaging.ml"; "src/client/client_protocol.ml";"src/client/client_main.ml";"src/nursery/nursery_main.ml" ]
```
Comment (NicolasT) [inline myocamlbuild.ml]: Could you put each file on a different line, a la

``` ocaml
let afpm = [
  "foo.ml";
  "bar.ml"
]
```

Having them all on a single line makes the line way too long to have any overview.

File: src/node/catchup.ml [hunk 2]
```diff
@@ -19,7 +19,7 @@ limitations under the License.
 open Lwt
 open Node_cfg
 open Remote_nodestream
-open Tlogcollection
+open Tlogcollection
```
Comment (NicolasT) [inline src/node/catchup.ml]: Superfluous space

File: src/node/statistics.ml [hunk 3]
```diff
@@ -341,18 +341,21 @@ module Statistics = struct
 
     let extract_list = function
       | Llio.NAMED_VALUELIST (_,l) -> l
-      | _ -> failwith "Wrong value type (expected list)"
+      | Llio.NAMED_FLOAT _ | Llio.NAMED_INT _ | Llio.NAMED_INT64 _ | Llio.NAMED_STRING _ 
+      -> failwith "Wrong value type (expected list)"
```
Comment (NicolasT) [inline src/node/statistics.ml]: Keep the arrow on the line of the match clauses (same remark applies to some occurrences below as well).

File: src/client/client_protocol.ml [hunk 4]
```diff
@@ -124,7 +124,22 @@ let decode_sequence ic =
...
+      | Update.SetRouting _
+      | Update.SetRoutingDelta _
+      | Update.Nop
+      | Update.Assert _
+      | Update.Assert_exists _
+      | Update.UserFunction _
+      | Update.AdminSet _
+      | Update.DeletePrefix _
+      | Update.Replace _ 
+        ->  raise (XException (Arakoon_exc.E_UNKNOWN_FAILURE,
```
Comment (NicolasT) [inline src/client/client_protocol.ml]: Keep the arrow on the line of the match(es).

File: src/node/store.ml [hunk 5]
```diff
@@ -680,7 +679,20 @@ struct
...
+      | Update.SetRouting _
+      | Update.SetRoutingDelta _
+      | Update.Nop
+      | Update.Assert _
+      | Update.Assert_exists _
+      | Update.UserFunction _
+      | Update.AdminSet _
+      | Update.DeletePrefix _
+      | Update.Replace _ 
+       -> None
```
Comment (NicolasT) [inline src/node/store.ml]: Keep the arrow on the line of the match(es).

File: src/paxos/multi_paxos_fsm.ml [hunk 6]
```diff
@@ -490,11 +486,20 @@ let wait_for_accepteds
                     Fsm.return (Slave_discovered_other_master new_state)
                   end
                 end
-            | Accept (n',i',_) (* when i' < i || (i' = i && n' < n) *) ->
+            (*Original 1.7:| Accept (n',i',_) (* when i' < i || (i' = i && n' < n) *) ->
```
Comment (NicolasT) [inline src/paxos/multi_paxos_fsm.ml]: Remove the commented section. It's confusing, especially due to nested comments.

File: src/paxos/multi_paxos_test.ml [hunk 7]
```diff
@@ -145,8 +156,19 @@ let test_generic network_factory n_nodes () =
       Logger.debug_f_ "%s: c0 from %s to %s" me (Multi_paxos_type.show_transition prev_key)
         (Multi_paxos_type.show_transition key) >>= fun () ->
       match key with
-        | (Multi_paxos_type.Stable_master x) -> Lwt.return (Some x)
-        | _ -> Lwt.return None
+          | (Multi_paxos_type.Stable_master x) -> Lwt.return (Some x)
```
Comment (NicolasT) [inline src/paxos/multi_paxos_test.ml]: Indentation

File: src/paxos/multi_paxos_test.ml [hunk 8]
```diff
@@ -174,9 +196,9 @@ let test_generic network_factory n_nodes () =
   let len = Hashtbl.length values in
   Logger.debug_f_ "%s: end of main... validating len = %d" me len >>= fun () ->
   let all_consensusses = Hashtbl.fold
-                           (fun a b acc ->
-                              let bs = Value.value2s b in
-                              (a,bs) :: acc) values []
+   (fun a b acc ->
```
Comment (NicolasT) [inline src/paxos/multi_paxos_test.ml]: Indentation.

File: src/paxos/multi_paxos_test.ml [hunk 9]
```diff
@@ -270,13 +292,24 @@ let test_master_loop network_factory ()  =
...
         (Multi_paxos_type.show_transition key) >>= fun () ->
       match key with
-        | (Multi_paxos_type.Stable_master x) ->
-          if !continue = 0
-          then Lwt.return (Some x)
-          else
-            let () = continue := (!continue -1) in
-            Lwt.return None
-        | _ -> Lwt.return None
+          | (Multi_paxos_type.Stable_master x) ->
```
Comment (NicolasT) [inline src/paxos/multi_paxos_test.ml]: Indentation.

File: src/paxos/multi_paxos_test.ml [hunk 10]
```diff
@@ -397,8 +429,19 @@ let test_simulation filters () =
       Logger.debug_f_ "%s: c0 from %s to %s" me (Multi_paxos_type.show_transition prev_key)
         (Multi_paxos_type.show_transition key) >>= fun () ->
       match key with
-        | (Multi_paxos_type.Stable_master x) -> Lwt.return (Some x)
-        | _ -> Lwt.return None
+          | (Multi_paxos_type.Stable_master x) -> Lwt.return (Some x)
```
Comment (NicolasT) [inline src/paxos/multi_paxos_test.ml]: Indentation

File: src/paxos/multi_paxos_test.ml [hunk 11]
```diff
@@ -427,8 +470,19 @@ let test_simulation filters () =
         (Multi_paxos_type.show_transition prev_key)
         (Multi_paxos_type.show_transition key) >>= fun () ->
       match key with
-        | (Multi_paxos_type.Slave_steady_state x) -> Lwt.return (Some x)
-        | _ -> Lwt.return None
+          | (Multi_paxos_type.Slave_steady_state x) -> Lwt.return (Some x)
```
Comment (NicolasT) [inline src/paxos/multi_paxos_test.ml]: Indentation.

File: src/paxos/slave.ml [hunk 12]
```diff
@@ -85,7 +85,7 @@ let slave_steady_state (type s) constants state event =
       end
     else if n' < n || i' < i
     then
-      begin
+       begin
```
Comment (NicolasT) [inline src/paxos/slave.ml]: Indentation.

File: src/tlog/tlogcollection_test.ml [hunk 13]
```diff
@@ -44,7 +44,8 @@ let setup factory test_name () =
               match status with
                 | Unix.WEXITED rc when rc = 0 -> make_dir dir
                 | Unix.WEXITED rc             -> Llio.lwt_failfmt "rm -rf '%s' gave rc %i" dir rc
-                | _                           -> Llio.lwt_failfmt "rm -rf '%s' failed" dir
+                | Unix.WSIGNALED _ | Unix.WSTOPPED _
+                 -> Llio.lwt_failfmt "rm -rf '%s' failed" dir
```
Comment (NicolasT) [inline src/tlog/tlogcollection_test.ml]: Arrow location.

File: src/tlog/tlogreader2.ml [hunk 14]
```diff
@@ -284,7 +285,7 @@ module O = struct (* correct but slow folder for .tlc (aka Old) format *)
   let _fold
         ~inflate
         ic ~index (lowerI:Sn.t) (too_far_i:Sn.t option) ~first a0 f =
-    ignore index;
+    let () = ignore index in
```
Comment (NicolasT) [inline src/tlog/tlogreader2.ml]: Spaces at end of line?

File: src/tlog/update.ml [hunk 15]
```diff
@@ -229,6 +227,21 @@ module Update = struct
...
+    | SetRouting _
+    | SetRoutingDelta _
+    | Nop
+    | Assert _
+    | Assert_exists _
+    | UserFunction _
+    | AdminSet _
+    | DeletePrefix _
+    | Replace _ 
+    -> false
```
Comment (NicolasT) [inline src/tlog/update.ml]: Arrow location.

File: src/paxos/multi_paxos_fsm.ml [hunk 16]
```diff
@@ -490,11 +486,20 @@ let wait_for_accepteds
...
-                Fsm.return (Wait_for_accepteds state)
-              end
+                Fsm.return (Wait_for_accepteds state)a
+              endxi*)
+              | Accept (n',i',_v) when i' < i || (i' = i && n' < n)  ->
+	            begin
+	              Logger.debug_f_ "%s: wait_for_accepted: dropping old Accept %S" me (string_of msg) >>= fun () ->
+	              Fsm.return (Wait_for_accepteds state)
+	            end
+            | Accept (n',i',_v') ->
```
Comment (domsj) [inline src/paxos/multi_paxos_fsm.ml:494]: Indentation

File: src/msg/tcp_messaging_test.ml [hunk 17]
```diff
@@ -28,11 +28,11 @@ class player id (m:messaging) =
     method _send target msg =
       Logger.debug_f_ "%s sending message %s to %s"
         id (string_of msg) target >>= fun () ->
-      m # send_message msg id target
+      m # send_message msg ~source:id ~target:target
 
     method serve ?(n=100) ?(lowest=0) opp =
       let msg = make_msg "ping" n in
-      m # send_message msg "a" opp >>=
+      m # send_message msg ~source:"a"~target:opp >>=
```
Comment (NicolasT) [inline src/msg/tcp_messaging_test.ml]: Heh, it's interesting this compiles/works. I'd think it's a bug in the OCaml grammar.

File: src/node/node_main.ml [hunk 18]
```diff
@@ -609,8 +609,12 @@ let _main_2 (type s)
...
                        fc),
                   "client_buffer"
-                | _ ->
-                  (fun () -> Lwt_buffer.add e inject_buffer), "inject"
+                | Multi_paxos.FromNode _    
+                | Multi_paxos.LeaseExpired _ 
+                | Multi_paxos.Quiesce  _
+                | Multi_paxos.Unquiesce 
+                | Multi_paxos.DropMaster _ 
+                ->  (fun () -> Lwt_buffer.add e inject_buffer), "inject"
```
Comment (NicolasT) [inline src/node/node_main.ml]: Missed this arrow?

File: src/paxos/multi_paxos_test.ml [hunk 19]
```diff
@@ -272,11 +294,22 @@ let test_master_loop network_factory ()  =
       match key with
         | (Multi_paxos_type.Stable_master x) ->
           if !continue = 0
-          then Lwt.return (Some x)
+            then Lwt.return (Some x)
```
Comment (NicolasT) [inline src/paxos/multi_paxos_test.ml]: Indentation. It's either

``` ocaml
if foo
  then bar
  else baz
```

or

``` ocaml
if foo
then bar
else baz
```

But not

``` ocaml
if foo
  then bar
else baz
```

:wink:

**Comment by NicolasT** [comment]
Thanks for handling the remarks. Some minor leftover nits, could you handle these as well?

I'll start CI for the current HEAD. If that passes and the comments above are addressed, LGTM.

**Comment by NicolasT** [comment]
`arakoon-git-launcher` 344, using `PR-466` (6fc951b).

**Comment by NicolasT** [comment]
CI passed except `right.system_tests_anomaly.test_disk_full_on_slave`.

**Pull request merged**